objects/src/accounts/storage/slot.rs

```diff
@@ -1,175 +1,48 @@
-use alloc::string::{String, ToString};
-
-use miden_crypto::merkle::SimpleSmt;
 use vm_core::{
     utils::{Deserializable, Serializable},
     Word, EMPTY_WORD,
 };
-use vm_processor::DeserializationError;
-
-use super::{map::EMPTY_STORAGE_MAP_ROOT, Felt, STORAGE_TREE_DEPTH};
-
-// CONSTANTS
-// ================================================================================================
-
-const MAX_VALUE_ARITY: u8 = u8::MAX - 1;
-
-const MIN_ARRAY_DEPTH: u8 = 2;
-const MAX_ARRAY_DEPTH: u8 = 64;
 
-const DEFAULT_SLOT_TYPE: StorageSlotType = StorageSlotType::Value { value_arity: 0 };
+use super::{map::EMPTY_STORAGE_MAP_ROOT, StorageMap};
 
-// STORAGE SLOT TYPE
+// STORAGE SLOT
 // ================================================================================================
 
-/// An object that represents the type of a storage slot.
+/// An enum that represents the type of a storage slot and it's value.
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub enum StorageSlotType {
-    /// Represents a slot that contains a value with the specified arity.
-    Value { value_arity: u8 },
-    /// Represents a slot that contains a commitment to a map with values with the specified arity.
-    Map { value_arity: u8 },
-    /// Represents a slot that contains a commitment to an array with capacity 2^depth with values
-    /// with the specified arity.
-    Array { depth: u8, value_arity: u8 },
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum StorageSlot {
+    Value(Word),
+    Map(StorageMap),
 }
 
-impl StorageSlotType {
-    /// Returns true if this storage slot type is valid.
-    ///
-    /// Valid storage slot types are defined as follows:
-    /// - value arity must be between 0 and 254 (inclusive).
-    /// - for Array types, depth must be between 2 and 64 (inclusive).
-    pub fn is_valid(&self) -> bool {
-        match self {
-            StorageSlotType::Value { value_arity } => *value_arity <= MAX_VALUE_ARITY,
-            StorageSlotType::Map { value_arity } => *value_arity <= MAX_VALUE_ARITY,
-            StorageSlotType::Array { depth, value_arity } => {
-                *value_arity < MAX_VALUE_ARITY
-                    && *depth >= MIN_ARRAY_DEPTH
-                    && *depth <= MAX_ARRAY_DEPTH
-            },
-        }
-    }
-
-    /// Returns true if this storage slot type is a value type with arity 0.
-    pub fn is_default(&self) -> bool {
-        match self {
-            StorageSlotType::Value { value_arity } => *value_arity == 0,
-            _ => false,
-        }
-    }
+impl StorageSlot {
```

Comment:
We also probably need something like

- `StorageSlot::get_type() -> StorageSlotType`
- `StorageSlot::to_word() -> Word`. This would return the value for simple value slots, and the root for maps.